### PR TITLE
Gallery store/fetch/delete embedded readmes file from flat container

### PR DESCRIPTION
**Pull request by lyndaidaii**
* Add new service to store/fetch/delete readme from flat container
* Modified logic to upload/display embedded readme

Addresses:https://github.com/NuGet/Engineering/issues/3593

### CODE REVIEW

File: src/NuGetGallery.Core/Services/CoreReadmeFileService.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            _metadata = metadata ?? throw new ArgumentNullException(nameof(metadata));
+        }
+
+        /// <summary>
+        /// Saves the package readme.md file to storage. This method should throw if the package
+        /// does not have an embedded readme file 
+        /// </summary>
+        /// <param name="package">The package associated with the readme.</param>
+        /// <param name="readmeFile">The content of readme file.</param>
+        public Task SaveReadmeFileAsync(Package package, Stream readmeFile)
```
Comment (lyndaidaii) [inline src/NuGetGallery.Core/Services/CoreReadmeFileService.cs:32]: Moved from https://github.com/NuGet/NuGetGallery/blob/dev/src/NuGetGallery/Services/PackageFileService.cs#L63

File: src/NuGetGallery.Core/Services/CoreReadmeFileService.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            return _fileStorageService.SaveFileAsync(_metadata.PackageContentFolderName, fileName, readmeFile, overwrite: true);
+        }
+
+        /// <summary>
+        /// Save the readme file from package stream. This method should throw if the package
+        /// does not have an embedded readme file 
+        /// </summary>
+        /// <param name="package">Package information.</param>
+        /// <param name="packageStream">Package stream with .nupkg contents.</param>
+        public async Task ExtractAndSaveReadmeFileAsync(Package package, Stream packageStream)
```
Comment (lyndaidaii) [inline src/NuGetGallery.Core/Services/CoreReadmeFileService.cs:60]: Moved from  https://github.com/NuGet/NuGetGallery/blob/dev/src/NuGetGallery/Services/PackageFileService.cs#L84

File: src/NuGetGallery.Core/Services/CoreReadmeFileService.cs (hunk 1 above)
Comment (joelverhagen) [inline src/NuGetGallery.Core/Services/CoreReadmeFileService.cs:32]: This shouldn't be public, right? I only see `ExtractAndSaveReadmeFileAsync` using this.

File: src/NuGetGallery.Core/Services/ICoreReadmeFileService.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+using NuGet.Services.Entities;
+
+namespace NuGetGallery
+{
+    /// <summary>
+    /// Provides readmes file related operations
+    /// </summary>
+    public interface ICoreReadmeFileService
+    {
+
```
Comment (joelverhagen) [inline src/NuGetGallery.Core/Services/ICoreReadmeFileService.cs:15]: nit: extra space

File: src/NuGetGallery.Core/Services/CoreReadmeFileService.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            return _fileStorageService.DeleteFileAsync(_metadata.PackageContentFolderName, fileName);
+        }
+
+        private string ReadmePathTemplate => $"{_metadata.PackageContentPathTemplate}/{ReadmeFileName}";
+
+        private string BuildReadmeFileName(Package package)
+            => FileNameHelper.BuildFileName(package, ReadmePathTemplate, string.Empty);
+
+        private string BuildReadmeFileName(string id, string version)
+            => FileNameHelper.BuildFileName(id, version, ReadmePathTemplate, string.Empty);
```
Comment (joelverhagen) [inline src/NuGetGallery.Core/Services/CoreReadmeFileService.cs:146]: Looks like `version` gets normalized in `FileNameHelper.BuildFileName`. Please confirm (with UTs and even manually) but we should not double normalize versions

**Review by joelverhagen — APPROVED** [review]
Please manually verify legacy scenarios to ensure no regression, but this looks good. :shipit:

File: src/NuGetGallery.Core/Services/CoreReadmeFileService.cs (hunk 4 above)
Comment (lyndaidaii) [reply src/NuGetGallery.Core/Services/CoreReadmeFileService.cs:146]: Verified that no double normalize versions here.
Similar to legacy readme,  their `versions ` are normalized in `FileNameHelper.BuildFileName `https://github.com/NuGet/NuGetGallery/blob/master/src/NuGetGallery/Services/PackageFileService.cs#L70

**Comment by lyndaidaii** [comment]
Manually verified changes along with orchestrator, and v3 pipeline

**Pull request merged**